source/core_mqtt.c

```diff
@@ -1763,6 +1763,8 @@ static MQTTStatus_t receiveSingleIteration( MQTTContext_t * pContext,
         ( void ) memmove( pContext->networkBuffer.pBuffer,
                           &( pContext->networkBuffer.pBuffer[ totalMQTTPacketLength ] ),
                           pContext->index );
+
+        pContext->lastPacketRxTime = pContext->getTime();
```

Comment:
```suggestion
        if( status == MQTTSuccess )
        {
            pContext->lastPacketRxTime = pContext->getTime();
        }
```

Reply:
Seems prudent to update lastPacketRxTime if and only if the incoming packet was parsed successfully.

Reply:
Thanks for your suggestion. I think, it doesn't matter if the packet is a bad response or anything not expected. The only thing to be considered here is if a packet received from the broker or not since basic validation is already done above.

Am I missing something?

Reply:
MQTT_ProcessIncomingPacketTypeAndLength checks that the id and remaining length fields are valid but does not validate the remainder of the packet. 

Since the function knows if the rest of the packet can be parsed, it might as well use that data to determine if a packet has been received.

Reply:
I got your point about the additional information. I'm still not sure if it would be really helpful though.

My whole point is if a packet received from the broker no matter what the packet is. Because, for example, even if `MQTTRecvFailed` by mismatching QoS, it should be considered received, I think. so preventing keepalive to be triggered everytime.

I'm not convinced myself either what is better. So please feel free to commit your suggestion and merge.

Reply:
Hello @onkwon, thanks for your response.

I am inclined to agree with @paulbartell. I think we should only update the last keep alive time if the packet is parsed correctly. Consider it "defensive programming".

Do you mind making the change or would you like us to do it for you?

Reply:
Thanks for your response. I made the change.